### ISSUE TITLE
 SerializationUtils.saveObject(Object, File f) fails in current master (3.9-SNAPSHOT)

### ISSUE THREAD

**Issue by rohanar** [deeplearning4j/deeplearning4j#1125]
Serializing ClusterSet object with SerializationUtils.saveObject(cs, new File(modelfilename));. This worked before updating to the latest in the mater branch. 

Error is Caused by: java.io.NotSerializableException: java.nio.HeapIntBuffer

**Comment by Zixxy** [comment]
Could you provide more data? I was trying to reproduce it but saveObject works for me on current master.

**Comment by agibsonccc** [comment]
Yeah it's fine now. The current master issue was related to BaseNDArray having a heap int buffer for storing shape information. I moved that away (underneath the covers) to being a databuffer which fixed it. Thanks for highlighting this.

**Comment by rohanar** [comment]
SerializationUtils.readObject(<serialized-filename>) fails.

Exception in thread "main" java.lang.RuntimeException: java.lang.IllegalStateException: unread block data
	at org.nd4j.linalg.util.SerializationUtils.readObject(SerializationUtils.java:40)

**Comment by Zixxy** [comment]
Seems working, I pushed the test that code passes.

**Comment by rohanar** [comment]
I will pass my code in a gist for you guys to run it at your end.

**Comment by rohanar** [comment]
I have played with this a bit and found the following: The problem occurs with serializing/deserializing bigger ClusterSet models. In a collection of 2525 documents, the file sizes of the serialized ParagraphVectors and ClusterSet models are 91,176KB and 20,077,440KB respectively. Serialized ClusterSet file size is unusually big. Deserializing fails when reading the ClusterSet from this file. All works fine with smaller files.
I did try removing the poiints before serializing the ClusterSet by ClusterSet.removePoints() method. This will not delete the cluster centers, and seems to be sufficient for using the model for finding the cluster of new points. When this was done, the deserialized file size came down to 199KB  and deserializing worked fine. It may be worth finding out why ClusterSet (with points not removed) is that big, and possibly find a smarter way of serializing ClusterSet models.
